Src/VimCore/CommonOperations.fs

```diff
@@ -2256,7 +2295,8 @@ type internal CommonOperations
     
     /// Ensure the view properties are met at the caret
     member x.EnsureAtCaret viewFlags = 
-        x.DoActionWhenReady (fun () -> x.EnsureAtPointSync x.CaretPoint viewFlags)
+        if _vimBufferData.CaretIndex = 0 then

```

Comment:
We do not need to consider things like z commands, because those are not commands that `ShouldRunForEachCaret` returns true for. If we say `z.` we mean "center the screen on the primary caret and ignore all secondary carets."

Consider using a caret on every line of a long file, say, a thousand lines long. It is impossible to keep all the carets on the screen at the same time so we only want to ensure view properties:

1. For the primary caret
2. After all actions have been performed

Reply:
Maybe a better way to describe it is that many Vim commands don't make sense to run for each caret or selection. For example, writing the buffer with `:w<CR>`. If we have five carets we don't want to write the buffer five times. Likewise, centering the screen on the primary caret with `z.` doesn't have a natural multiple-caret interpretation.